### PR TITLE
LPS-159735 | Add support for nestedFields query param to API Explorer

### PR DESCRIPTION
**Pull request by Hijky**
**Background**
Add support for nestedFields query param to API Explorer

**Test Plan**
Given a display page templates created
When with curl I request GET getSiteDisplayPageTemplatesPage with siteId and nestedFields = "profileURL"
Then in a response I receive a with profileURL field values in the creator field have appeared

Given an Asset Library with a Web Content Folder created
When with curl I request GET getAssetLibraryStructuredContentFoldersPage with assetLibraryId and nestedFields = "profileURL"
Then in a response I receive a with profileURL field values in the creator field have appeared

**JIRA Ticket**
https://issues.liferay.com/browse/LPS-159735

### CODE REVIEW

**Comment by Hijky** [comment]
ci:test:headless-frontend-functional:nocompile

File: portal-web/test/functional/com/liferay/portalweb/macros/headless/journal/HeadlessWebcontentFolderAPI.macro [hunk 1]
```diff
@@ -189,6 +189,25 @@ definition {
...
 
+	macro _getStructuredContentFoldersWithDifferentParameters {
+		var portalURL = JSONCompany.getPortalURL();
+		var api = "asset-libraries/${assetLibraryId}/structured-content-folders";
+
+		if (isSet(parameter)) {
+			var api = StringUtil.add("${api}", "?${parameter}=${parameterValue}", "");
+		}
+
+		var curl = '''
```
Comment (magjed4289) [inline portal-web/test/functional/com/liferay/portalweb/macros/headless/journal/HeadlessWebcontentFolderAPI.macro:200]: please format to get an alignment like this one:
image

**Review by magjed4289 — CHANGES_REQUESTED** [review]

File: portal-web/test/functional/com/liferay/portalweb/macros/headless/journal/HeadlessWebcontentFolderAPI.macro (hunk 1 above)
Comment (Hijky) [reply portal-web/test/functional/com/liferay/portalweb/macros/headless/journal/HeadlessWebcontentFolderAPI.macro:200]: I have updated it.

**Review by magjed4289 — APPROVED** [review]

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/headless/HeadlessDeliveryAPI/HeadlessDeliverySupportDifferentParameters.testcase [hunk 2]
```diff
@@ -0,0 +1,131 @@
+@component-name = "portal-headless-frontend-infrastructure"
```
Comment (yanan) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/headless/HeadlessDeliveryAPI/HeadlessDeliverySupportDifferentParameters.testcase:1]: Great to rename the file to including different parameters!

**Review by yanan — APPROVED** [review]

**Comment by Hijky** [comment]
ci:forward

**Pull request closed without merging**